compilers/concrete-compiler/compiler/lib/Common/Values.cpp

```diff
@@ -0,0 +1,280 @@
+using concretelang::protocol::Message;
+using concretelang::protocol::protoPayloadToVector;
+using concretelang::protocol::protoShapeToDimensions;
+using concretelang::protocol::vectorToProtoPayload;
+
+namespace concretelang {
+namespace values {
+
+Value Value::fromRawTransportValue(TransportValue transportVal) {
+  Value output;
+  auto integerPrecision =
+      transportVal.asReader().getRawInfo().getIntegerPrecision();
+  auto isSigned = transportVal.asReader().getRawInfo().getIsSigned();
+  auto dimensions =
+      protoShapeToDimensions(transportVal.asReader().getRawInfo().getShape());
+  auto data = transportVal.asReader().getPayload();
+  if (integerPrecision == 8 && isSigned) {
+    auto values = protoPayloadToVector<int8_t>(data);
+    output.inner = Tensor<int8_t>{values, dimensions};
+  } else if (integerPrecision == 16 && isSigned) {
+    auto values = protoPayloadToVector<int16_t>(data);
+    output.inner = Tensor<int16_t>{values, dimensions};
+  } else if (integerPrecision == 32 && isSigned) {
+    auto values = protoPayloadToVector<int32_t>(data);
+    output.inner = Tensor<int32_t>{values, dimensions};
+  } else if (integerPrecision == 64 && isSigned) {
+    auto values = protoPayloadToVector<int64_t>(data);
+    output.inner = Tensor<int64_t>{values, dimensions};
+  } else if (integerPrecision == 8 && !isSigned) {
+    auto values = protoPayloadToVector<uint8_t>(data);
+    output.inner = Tensor<uint8_t>{values, dimensions};
+  } else if (integerPrecision == 16 && !isSigned) {
+    auto values = protoPayloadToVector<uint16_t>(data);
+    output.inner = Tensor<uint16_t>{values, dimensions};
+  } else if (integerPrecision == 32 && !isSigned) {
+    auto values = protoPayloadToVector<uint32_t>(data);
+    output.inner = Tensor<uint32_t>{values, dimensions};
+  } else if (integerPrecision == 64 && !isSigned) {
+    auto values = protoPayloadToVector<uint64_t>(data);
+    output.inner = Tensor<uint64_t>{values, dimensions};
+  } else {
+    assert(false);
+  }
+
+  return output;
+}
+
+TransportValue Value::intoRawTransportValue() const {
+  auto output = Message<concreteprotocol::Value>();
+  auto rawInfo = output.asBuilder().initRawInfo();
+  rawInfo.setShape(intoProtoShape().asReader());
+  rawInfo.setIntegerPrecision(getIntegerPrecision());
+  rawInfo.setIsSigned(isSigned());
+  output.asBuilder().setPayload(intoProtoPayload().asReader());
+  return output;
+}
+
+uint32_t Value::getIntegerPrecision() const {
+  if (hasElementType<uint8_t>() || hasElementType<int8_t>()) {
+    return 8;
+  } else if (hasElementType<uint16_t>() || hasElementType<int16_t>()) {
+    return 16;
+  } else if (hasElementType<uint32_t>() || hasElementType<int32_t>()) {
+    return 32;
+  } else if (hasElementType<uint64_t>() || hasElementType<int64_t>()) {
+    return 64;
+  } else {
+    assert(false);
+  }
+}
+
+uint32_t Value::isSigned() const {
+  if (hasElementType<uint8_t>() || hasElementType<uint16_t>() ||
```

Comment:
Nit: This may be delegated to `std::is_signed`.

Reply:
That's much better yes !